### PR TITLE
Add layer config access in legend

### PR DESCRIPTION
**Pull request by hav142**
This PR is still drafted to fixes issue #793 

How to test the feature:

- activate any layer
- check the icons on the legend

### CODE REVIEW

**Comment by wadhwamatic** [comment]
Nice work on this @hav142! Some minor inputs from me which are just for consideration:
1. I think the icons can now be left aligned since they no longer are in-line with the layer name. What do you think?
2. A little more spacing between the legend detail text and the icons would be good to give some more room
3. I think we can space out the icons a bit too

**Comment by hav142** [comment]
@wadhwamatic Thanks for the suggestions. Is this good enough?

image

**Comment by wadhwamatic** [comment]
@hav142 - yes, that looks good. as you move forward with this, please have a look at https://github.com/WFP-VAM/prism-app/pull/920 to make sure that the CSV download feature for exposure analysis works with your revised UI

**Comment by ericboucher** [comment]
@hav142 can you have a look at the linting issues? The demo endpoint will only get deployed if all the tests and linters are passing. Thanks!

File: frontend/yarn.lock [hunk 1]
```diff
@@ -2690,14 +2690,28 @@
   dependencies:
     "@types/react" "*"
 
-"@types/react@*", "@types/react@^16.9.0":
+"@types/react@*":
   version "16.9.23"
```
Comment (ericboucher) [inline frontend/yarn.lock:2694]: Should we update the types too?

**Comment by wadhwamatic** [comment]
Thanks for picking this up again @echaidemenos. This works well except when I try to download a raster (WMS) layer. The download link works in the layer navigation, but not in the legend.

Screenshot 2023-11-08 at 13 33 35

@ericboucher - this sounds related to the issue we saw earlier today.

**Review by ericboucher — APPROVED** [review]

**Review by wadhwamatic — APPROVED** [review]
Looks good, thanks @echaidemenos !

**Pull request merged**